posix_test.go

```diff
@@ -401,6 +405,159 @@ func TestDeleteFile(t *testing.T) {
 	}
 }
 
+// Test posix.ReadFile()
+func TestReadFile(t *testing.T) {
+	// Create temporary directory.
+	path, err := ioutil.TempDir("", "minio-")
+	if err != nil {
+		t.Fatalf("Unable to create a temporary directory, %s", err)
+	}
+	defer removeAll(path)
+
+	// Initialize posix storage layer.
+	posix, err := newPosix(path)
+	if err != nil {
+		t.Fatalf("Unable to initialize posix, %s", err)
+	}
+
+	// Setup test environment.
+	if err = posix.MakeVol("success-vol"); err != nil {
+		t.Fatalf("Unable to create volume, %s", err)
+	}
+
+	// Create directory to make errIsNotRegular
+	if err = os.Mkdir(slashpath.Join(path, "success-vol", "object-as-dir"), 0777); err != nil {
+		t.Fatalf("Unable to create directory, %s", err)
+	}
+
+	testCases := []struct {
+		fileName    string
+		offset      int64
+		bufSize     int
+		expectedBuf []byte
+		expectedErr error
+	}{
```

Comment:
It'll be good to add case numbers before the cases `// Test case  - n`, easy to follow in case of failures.